src/libbson/src/bson/bson-string.c

```diff
@@ -145,9 +182,9 @@ bson_string_free (bson_string_t *string, /* IN */
 /*
  *--------------------------------------------------------------------------
  *
- * bson_string_append --
+ * bson_string_append_ex --
  *
- *       Append the UTF-8 string @str to @string.
+ *       Append the UTF-8 string @str of given length @len to @string.
```

Comment:
The behavior when `len` > `strlen(str)` is undefined but doesn't crash. I don't think we want to add an assertion because that will require an `O(n)` check.

Reply:
It is enough for `len <= strlen(str)` to be an unasserted precondition of the function (as is currently written, "of given length `@len`"). A more featureful contracts API would allow for distinguishing "cheap" vs. "expensive" assertions, but we do not have such an API. It is fine to leave it to ASan to detect violations for now.